userlevel/click.cc

```diff
@@ -620,32 +623,41 @@ main(int argc, char **argv)
      case NO_WARNINGS_OPT:
       warnings = clp->negated;
       break;
-
+#if HAVE_DPDK
+     case DPDK_OPT: {
+      Vector<String> argset;
+      const char* arg = clp->vstr;
+      while (1) {
+          if (strcmp(arg, "--") == 0)
+              break;
+          argset.push_back(arg);
+          arg = Clp_Shift(clp, 1);
+      }
+      dpdk_argv = (char**)malloc(sizeof(char*) * argset.size() + 1);
+      dpdk_argv[0] = argv[0];
+      for (int i = 0; i < argset.size() ; i++) {
+        dpdk_argv[i+1] = const_cast<char*>(argset[i].c_str());
+      }
+      dpdk_enabled = true;
+      dpdk_argc = clp->have_val + 1;
+      break;
+     }
+#endif // HAVE_DPDK
```

Comment:
I don't think this'll do what you want, because going to & from `String` will end up referencing freed memory. Try this (assuming you turn off `Clp_ValString` as suggested above)

```
Vector<char*> dpdk_arg; // no argc,argv
...
#if HAVE_DPDK
case DPDK_OPT: {
    const char* arg;
    while ((arg = Clp_Shift(clp, 1)) != NULL && strcmp(arg, "--") != 0)
        dpdk_arg.push_back(const_cast<char*>(arg));
    dpdk_arg.push_back(NULL);
    dpdk_enabled = true;
    break;
}
#endif // HAVE_DPDK
```